### PR TITLE
METRON-1410 [MPACK] Check for existing HBASE tables before creating

### PR DESCRIPTION
**Pull request by ottobackwards**
After upgrade, indexing will not restart because it fails to create HBase tables.  This is because they already exist from the previous version.

This PR changes the base command used to check for the existence of the table, and only create it if it does not already exist.

## Testing
- Full Dev should deploy as normal
- [tbd] some upgrade procedure

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? If not one needs to be created at [Metron Jira](https://issues.apache.org/jira/browse/METRON/?selectedTab=com.atlassian.jira.jira-projects-plugin:summary-panel). 
- [x] Does your PR title start with METRON-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.
- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

### For code changes:
- [ ] Have you included steps to reproduce the behavior or problem that is being changed or addressed?
- [ ] Have you included steps or a guide to how the change may be verified and tested manually?
- [x] Have you ensured that the full suite of tests and checks have been executed in the root metron folder via:
  ```
  mvn -q clean integration-test install && build_utils/verify_licenses.sh 
  ```

- [ ] Have you written or updated unit tests and or integration tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [x] Have you verified the basic functionality of the build by building and running locally with Vagrant full-dev environment or the equivalent?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered by building and verifying the site-book? If not then run the following commands and the verify changes via `site-book/target/site/index.html`:

  ```
  cd site-book
  mvn site
  ```

### CODE REVIEW

File: metron-deployment/packaging/ambari/metron-mpack/src/main/resources/common-services/METRON/CURRENT/package/scripts/enrichment_commands.py [hunk 1]
```diff
@@ -190,7 +190,7 @@ def create_hbase_tables(self):
                   self.__params.hbase_principal_name,
                   execute_user=self.__params.hbase_user)
 
-        cmd = "echo \"create '{0}','{1}'\" | hbase shell -n"
+        cmd = "if [[ $(echo \"exists '{0}'\" | hbase shell | grep 'not exist') ]]; then echo \"create '{0}','{1}'\" | hbase shell -n; fi"
```
Comment (nickwallen) [inline metron-deployment/packaging/ambari/metron-mpack/src/main/resources/common-services/METRON/CURRENT/package/scripts/enrichment_commands.py]: Would it be possible to just add a `create_hbase_table` function in `metron_service` that is used by Enrichment, Indexing, and the Profiler?  Then we don't have to duplicate this in three places?

Comment (ottobackwards) [reply metron-deployment/packaging/ambari/metron-mpack/src/main/resources/common-services/METRON/CURRENT/package/scripts/enrichment_commands.py]: I thought of that, the issue would be that that function would have the same parameters as this Execute command.  It wouldn't really make it better.

We could create a shared get_create_hbase_table_fmt() function that returns the command.

Comment (ottobackwards) [reply metron-deployment/packaging/ambari/metron-mpack/src/main/resources/common-services/METRON/CURRENT/package/scripts/enrichment_commands.py]: That statement is based on us not always possibly wanting to use the same users etc.

Comment (cestella) [reply metron-deployment/packaging/ambari/metron-mpack/src/main/resources/common-services/METRON/CURRENT/package/scripts/enrichment_commands.py]: Still seems like not having to cut and paste that bit of shell script inside a python variable is worth generalizing to a `create_hbase_table` function.

Comment (ottobackwards) [reply metron-deployment/packaging/ambari/metron-mpack/src/main/resources/common-services/METRON/CURRENT/package/scripts/enrichment_commands.py]: ok, i'll do that

Comment (ottobackwards) [reply metron-deployment/packaging/ambari/metron-mpack/src/main/resources/common-services/METRON/CURRENT/package/scripts/enrichment_commands.py]: done

**Comment by nickwallen** [comment]
+1 Looks good @ottobackwards.  Thanks for fixing this!

I have not tested this myself, but it looks solid.  Let me know if you'd prefer me to spin this up to get a second test run in.  Otherwise, I assume that what you've done is sufficient.

**Comment by ottobackwards** [comment]
I'm going to wait for @lvets a chance to try his scenario

**Comment by mmiklavc** [comment]
Thanks for fixing this @ottobackwards. +1 via inspection from me as well.

**Pull request closed without merging**